core/src/main/java/org/apache/iceberg/rest/BaseHTTPClient.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest;
+
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import org.apache.iceberg.rest.HTTPRequest.HTTPMethod;
+import org.apache.iceberg.rest.auth.AuthSession;
+import org.apache.iceberg.rest.responses.ErrorResponse;
+
+/** A base class for {@link RESTClient} implementations using {@link HTTPRequest}. */
+public abstract class BaseHTTPClient implements RESTClient {
```

Comment:
It's more than just a refactoring, `BaseHTTPClient` is playing an important role here: it's "bridging" the existing `RESTClient` methods (`get()`, `post()`, `postForm()` etc.) towards a single target: the `buildRequest()` method, which returns an `HTTPRequest` instance. Without this class, each implementation of `get()`, `post()`, `postForm()` etc. would need to do the job of creating an `HTTPRequest` then passing it to the authenticator, then executing it – it would be a lot more confusing.

Reply:
it still adds too much reviewing complexity to an already large PR, hence why I'm suggesting to do this in a separate PR

Reply:
Okay, `BaseHTTPClient` has been removed.